context/AuthContext.js

```diff
@@ -1,19 +1,57 @@
-import React, { children, createContext, useState } from "react";
+import React, { createContext, useState, useEffect, useCallback } from "react";
+import AsyncStorage from "@react-native-async-storage/async-storage";
 export const AuthContext = createContext();
 export const AuthProvider = ({ children }) => {
-  const [isLoding, setIsLoading] = useState(true);
+  const [isLoading, setIsLoading] = useState(true);
   const [userToken, setUserToken] = useState(null);
-  const authFinish = (token) => {
-    setUserToken(token);
+
+  const authFinish = (userUid) => {
+    setUserToken(userUid);
     setIsLoading(false);
-    console.log(token);
+    storeUserToken(userUid);
   };
+
+  const retrieveData = useCallback(async () => {
+    try {
+      const value = await AsyncStorage.getItem("userToken");
+      if (value !== null) {
+        setUserToken(value);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }, []);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      await retrieveData();
+    };
+    fetchData();
```

Comment:
you can just call retreiveData() directly here 😅 
No need for fetchData function